cmds/chroot/chroot.go

```diff
@@ -0,0 +1,191 @@
+		"the working directory to / after changing the root directory to newroot, i.e., ",
+		"inside the chroot. This option is only permitted when newroot is the old / directory."))
+	flag.Parse()
+}
+
+func stringToUint32(str string) (uint32, error) {
+	ret, err := strconv.ParseUint(str, 10, 32)
+	if err != nil {
+		return 0, err
+	}
+	return uint32(ret), nil
+}
+
+func getCurrentUser() (uid uint32, gid uint32) {
+	return uint32(syscall.Geteuid()), uint32(syscall.Getegid())
+}
+
+func parseUserspec() (uid uint32, gid uint32, err error) {
+	userspecSplit := strings.Split(userspecFlag, ":")
+	if len(userspecSplit) != 2 || userspecSplit[1] == "" {
+		return 0, 0, fmt.Errorf("Expected user spec flag to be \":\" separated values received %s", userspecFlag)
+	}
+
+	uid, err = stringToUint32(userspecSplit[0])
+	if err != nil {
+		return 0, 0, err
+	}
+
+	gid, err = stringToUint32(userspecSplit[1])
+	if err != nil {
+		return 0, 0, err
+	}
+
+	return uid, gid, nil
+}
+
+func parseGroups() (groups []uint32, err error) {
+	groupStrs := strings.Split(groupsFlag, ",")
+	ret := make([]uint32, len(groupStrs))
+	for index, group := range groupStrs {
+
+		gid, err := stringToUint32(group)
+		if err != nil {
+			return ret, err
+		}
+
+		ret[index] = gid
+	}
+	return ret, nil
+}
+
+func parseCommand(args []string) []string {
+	if len(args) > 1 {
+		return args[1:]
+	}
+	return []string{"/bin/sh", "-i"}
+}
+
+func parseRoot(args []string) (root string, err error) {
+	if len(args) < 1 {
+		return "", fmt.Errorf("Missing operand")
+	}
+
+	return filepath.Abs(args[0])
+}
+
+func isRoot(dir string) (bool, error) {
+	realPath, err := filepath.EvalSymlinks(dir)
+	if err != nil {
+		return false, err
+	}
+	absolutePath, err := filepath.Abs(realPath)
+	if err != nil {
+		return false, err
+	}
+	if absolutePath == "/" {
+		return true, nil
+	}
+	return false, nil
+}
+
+func setIDs(uid *uint32, gid *uint32, extraGids *[]uint32) (err error) {
+	var (
+		tuid       uint32
+		tgid       uint32
+		textraGids []uint32
+	)
+
+	if userspecFlag == "" {
+		tuid, tgid = getCurrentUser()
+	} else {
+		tuid, tgid, err = parseUserspec()
+		if err != nil {
+			return err
+		}
+	}
+
+	if groupsFlag != "" {
+		textraGids, err = parseGroups()
+		if err != nil {
+			return err
+		}
+	}
+
+	*uid = tuid
+	*gid = tgid
+	*extraGids = textraGids
+	return nil
+}
+
+func main() {
+	var (
+		uid       uint32
+		gid       uint32
+		extraGids []uint32
+		newRoot   string
+		isOldroot bool
+		err       error
+	)
+
+	if flag.NFlag() == 0 && flag.NArg() == 0 {
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
+
+	newRoot, err = parseRoot(flag.Args())
+	if err != nil {
+		log.Fatal(err)
+	}
+	isOldroot, err = isRoot(newRoot)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if !skipchdirFlag {
+		err = os.Chdir(newRoot)
+		if err != nil {
+			log.Fatal(err)
+		}
+	} else if !isOldroot {
+		log.Fatal(fmt.Errorf("The -s option is only permitted when newroot is the old / directory"))
```

Comment:
No need to use fmt.Errorf here, this can just be log.Fatal (or if you did have format strings, log.Fatalf)